### PR TITLE
Fix HttpProxyService Disposal and Logs Causing Teamcity to Hang

### PR DESCRIPTION
**Pull request by nathanwoctopusdeploy**
# Background

Fix HttpProxyService Disposal
Reduced logging level for Halibut is some noisy tests as this was causing the Windows Test Build to hang

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Halibut.TestProxy/ProxyEndpoint.cs [hunk 1]
```diff
@@ -3,26 +3,5 @@
 
 namespace Halibut.TestProxy
 {
-    record ProxyEndpoint(string Hostname, int Port)
-    {
-        static Regex ParseRegex = new Regex("(?<hostname>[\\S]+):(?<port>\\d+)", RegexOptions.Compiled);
```
Comment (nathanwoctopusdeploy) [inline source/Halibut.TestProxy/ProxyEndpoint.cs:8]: This code was unused

File: .nuke/build.schema.json [hunk 2]
```diff
@@ -33,6 +33,7 @@
             "AppVeyor",
             "AzurePipelines",
             "Bamboo",
+            "Bitbucket",
```
Comment (nathanwoctopusdeploy) [inline .nuke/build.schema.json:36]: I ran the tests through nuke and it did this 🤷

File: source/Halibut.Tests/FailureModesFixture.cs [hunk 3]
```diff
@@ -138,6 +139,8 @@ public async Task FailWhenServerThrowsDuringADataStream(ServiceConnectionType se
                        .ForServiceConnectionType(serviceConnectionType)
                        .WithReadDataStreamService()
                        .WithPollingReconnectRetryPolicy(() => new RetryPolicy(1, TimeSpan.Zero, TimeSpan.Zero))
+                       // This test is very noisy for the test logs
+                       .WithHalibutLoggingLevel(LogLevel.Fatal)
```
Comment (nathanwoctopusdeploy) [inline source/Halibut.Tests/FailureModesFixture.cs:143]: Some of the tests are generating 1000s of Halibut log messages and this was killing the WIndows Test Build on Teamcity.

File: .nuke/build.schema.json (hunk 2 above)
Comment (LukeButters) [reply .nuke/build.schema.json:36]: Does this for me as well.

File: source/Halibut.TestUtils.CompatBinary.Base/SettingsHelper.cs [hunk 4]
```diff
@@ -46,30 +47,40 @@ public static ServiceConnectionType GetServiceConnectionType()
...
+            {
+                return (LogLevel)Enum.Parse(typeof(LogLevel), s);
+            }
+        }
+
         public static X509Certificate2 GetClientCertificate()
         {
             var octopusCertPath = GetSetting("octopuscertpath");
-            Console.WriteLine($"Using octopus cert path: {octopusCertPath}");
+            //Console.WriteLine($"Using octopus cert path: {octopusCertPath}");
```
Comment (LukeButters) [inline source/Halibut.TestUtils.CompatBinary.Base/SettingsHelper.cs:63]: I think these can just be deleted, eventually.

**Review by LukeButters — APPROVED** [review]

**Pull request merged**